scp-dbus-service.py

```diff
@@ -162,6 +163,26 @@ def _disconnect_signals (self):
         for s in self._signals:
             g_ppds.disconnect (s)
 
+    def _needs_querying(self, ppdname):
+        # No need to query if we already tried downloading a driver
+        if self.download_tried:
+            return False
+
+        # In EOS we force querying OpenPrinting.org even if an 'exact' fit is
+        # available, unless it's a driver that we already downloaded before via
+        # eos-config-printer, or one of those we chose to pre-install in the OS.
+        regexps = [
+            'lsb/.*eos-config-printer.*ppd.*',   # automatically downloaded via e-c-p
+            'lsb/.*OpenPrinting.*ppd.*',         # e.g. openprinting-ppds-postscript-brother
+            'lsb/.*epson-inkjet-printer.*ppd.*', # e.g. epson-inkjet-printer-201401w
+        ]
```

Comment:
Not a huge fan of this hardcoded list of strings tbh. What makes the previous logic conclude that there is an `exact` fit? Could you introduce another fit that is stronger than `exact` and move the heuristic elsewhere? (Sorry if this does not make sense given the architecture of s-c-p, I'm not too familiar with the internals.)